syntax = "proto3";
package pb;

service TransactionService {
    rpc FetchTransactions (FetchRequest) returns (Transactions) {}
}

message Transaction {
    int32 account_id = 1;
    float amount = 2;
    int32 transaction_date = 3;
    string description = 4;
    float fx_rate = 5;
    
    enum AccountType {
        BANK = 0;
        CREDIT_CARD = 1;
    }

    AccountType account_type = 6;
}

message Transactions {
    repeated Transaction transactions = 1;
}

message FetchRequest {

}
